src/utils.cpp

```diff
@@ -1481,4 +1481,30 @@ std::vector<int> get_all_running_pids()
   return pids;
 }
 
+// BPF verifier rejects programs with names containing certain characters, use
+// this function to replace every character not valid for C identifiers by '_'
+std::string sanitiseBPFProgramName(std::string const &name)
```

Comment:
Threw me off a bit - thought it was like difference between `const char *` and `char *const`. Recommend sticking to `const std::string &` like rest of codebase

Reply:
Also would be good to have a few unit tests for this, if only to document expected input and output

Reply:
Thanks! Fixed and added a unit test.